### ISSUE TITLE
False positive: SlevomatCodingStandard.Namespaces.UnusedUses.MismatchingCaseSensitivity (with searchAnnotations)

### ISSUE THREAD

**Issue by LastDragon-ru** [slevomat/coding-standard#1260]
```php
<?php declare(strict_types = 1);

namespace A {
    use B\DataProvider;

    class SpaControllerTest /* extends TestCase */ {
        /**
         * @dataProvider dataProviderSettings
         */
        public function testSettings(): void {
            // empty
        }

        public function dataProviderSettings(): DataProvider {
            return new class() implements DataProvider {
                /**
                 * @inheritDoc
                 */
                public function getData(bool $raw = false): array {
                    return [];
                }
            };
        }
    }
}

namespace B {
    interface DataProvider {
        // empty
    }
}

```

```
<?xml version="1.0"?>
<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <config name="installed_paths" value="../../slevomat/coding-standard"/>
    <file>./packages</file>
    <rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
        <properties>
            <property name="searchAnnotations" value="true"/>
        </properties>
    </rule>
</ruleset>
```

```
 ./vendor/bin/phpcs ./packages/Bug.php

FILE: /project/packages/Bug.php
----------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------
 8 | ERROR | Case of reference name "dataProvider" and use statement "DataProvider" does not match.
----------------------------------------------------------------------------------------------------

Time: 155ms; Memory: 10MB
```

```
name     : slevomat/coding-standard
descrip. : Slevomat Coding Standard for PHP_CodeSniffer complements Consistence Coding Standard by providing sniffs with additional checks.
keywords :
versions : * dev-master, * 7.x-dev
type     : phpcodesniffer-standard
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage :
source   : [git] https://github.com/slevomat/coding-standard.git db243812192ab2308098dfeec5ea8a95b93aba18
dist     : [zip] https://api.github.com/repos/slevomat/coding-standard/zipball/db243812192ab2308098dfeec5ea8a95b93aba18 db243812192ab2308098dfeec5ea8a95b93
a
```

![image](https://user-images.githubusercontent.com/1329824/128602051-ab7baef2-a51a-4bc7-8161-dfbb8810b22b.png)

**Comment by kukulich** [comment]
There are more options in the sniff.

**Comment by LastDragon-ru** [comment]
So in the cases like this, I just need to add it into `ignoredAnnotationNames`?

**Comment by kukulich** [comment]
@LastDragon-ru Yes.
